src/libYARP_sig/src/Image.cpp

```diff
@@ -637,7 +660,13 @@ Image::~Image() {
 
 
 int Image::getPixelSize() const {
-    return imgPixelSize;
+
+    if(pixelCode2Size.find((YarpVocabPixelTypesEnum)imgPixelCode) != pixelCode2Size.end())
+    {
+        return -imgPixelCode;
+    }
+
+    return pixelCode2Size.at((YarpVocabPixelTypesEnum)imgPixelCode);
```

Comment:
I don't like this...
The `getPixelSize` method should just return `imgPixelSize`. Instead this logic should be used when setting the pixelCode

Reply:
i think it is better to leave the getter method directly linked to the pixelCode value. for someone who read the code the binding between the two is clearer this way. The cached variable is used only in the getPixelAdress Loop.